src/Mod/Part/Gui/Command.cpp

```diff
@@ -1329,7 +1380,7 @@ CmdPartExtrude::CmdPartExtrude()
 void CmdPartExtrude::activated(int iMsg)
 {
     Q_UNUSED(iMsg);
-    Gui::Control().showDialog(new PartGui::TaskExtrusion());
+    PartGui::checkSkipDialog(new PartGui::TaskExtrusion(), "makeExtrude", "Extrude");

```

Comment:
Text used for Undo transactions are generally translated

Reply:
Not here:

https://github.com/FreeCAD/FreeCAD/blob/796efc4c6b76b93ea36905eb013b79bb197dbbef/src/Mod/Sketcher/Gui/TaskSketcherValidation.cpp#L161

https://github.com/FreeCAD/FreeCAD/blob/853454a13920788120234d5e0f64b172a9e27315/src/Mod/Mesh/Gui/Segmentation.cpp#L108

https://github.com/FreeCAD/FreeCAD/blob/853454a13920788120234d5e0f64b172a9e27315/src/Mod/Points/Gui/Command.cpp#L340

https://github.com/FreeCAD/FreeCAD/blob/5d49bf78de785a536f941f1a6d06d432582a95d3/src/Mod/Part/Gui/Mirroring.cpp#L135

https://github.com/FreeCAD/FreeCAD/blob/5d49bf78de785a536f941f1a6d06d432582a95d3/src/Mod/Part/Gui/DlgRevolution.cpp#L356

Reply:
You'll find untranslated `openTransaction` calls throughout the codebase because it didn't used to be supported (at least, that's my understanding of a discussion with @wwmayer at https://forum.freecadweb.org/viewtopic.php?f=35&t=65436). I believe it's now allowed to translate that text.

Reply:
Okay.  I will add it then.  QT_TR_NOOP() is used in other places in that file, so I will use that, too.

Reply:
I don't know if the NOOP variant is correct there: does FreeCAD internally call translate() on the Undo menu item? Honestly the undo system is still a bit of a mystery to me, we may have to just test it.

Reply:
I guess we should move the conversation to the newest PR.

https://github.com/FreeCAD/FreeCAD/pull/6380